### PR TITLE
New signup flow

### PR DESCRIPTION
**Pull request by dylanwh**
This introduces a much simplified signup flow, over email.

This code is live on https://harmony.bugzilla.ninja

## Before

1. Visit createaccount.cgi
2. Type email
3. Get email, click link
4. Create account
5. Re-enter login and password.

## After

1. Type email into any page
2. Get email, click link
3. Create account + login at the same time

### CODE REVIEW

File: Bugzilla/Util.pm [hunk 1]
```diff
@@ -329,15 +329,7 @@ sub do_ssl_redirect_if_required {
 
 # Returns the real remote address of the client,
 sub remote_ip {
-  if (($ENV{SERVER_SOFTWARE} // '') eq 'Bugzilla::App::CGI') {
```
Comment (dylanwh) [inline Bugzilla/Util.pm:332]: This is broken and needs to be extracted to another PR.

File: Bugzilla/App/Plugin/Glue.pm [hunk 2]
```diff
@@ -80,6 +80,16 @@ sub register {
 
       my $name = sprintf '%s.%s.tmpl', $options->{template}, $options->{format};
       my $template = Bugzilla->template;
+      if ($options->{variant}) {
```
Comment (dylanwh) [inline Bugzilla/App/Plugin/Glue.pm:83]: This should be in a different PR

**Comment by CyberShadow** [comment]
@dylanwh  There are still a bunch of remaining references / links to createaccount.cgi, which now 404.
